cli/command/node/cmd.go

```diff
@@ -38,6 +41,16 @@ func Reference(ctx context.Context, client apiclient.APIClient, ref string) (str
 		if err != nil {
 			return "", err
 		}
+		if info.Swarm.NodeID == "" {
+			// If there's no node ID in /info, the node probably
+			// isn't a manager. Call a swarm-specific endpoint to
+			// get a more specific error message.
+			_, err = client.NodeList(ctx, types.NodeListOptions{})
```

Comment:
It looks like we now have different approaches (and multiple locations) where we check if it's a manager;

using `info.Swarm.ControlAvailable`;
https://github.com/docker/docker/blob/b0d1936d304824dc0438b200d269e2d4e0751fcc/cli/command/stack/deploy.go#L65-L78

using `info.Swarm.LocalNodeState`;
https://github.com/docker/docker/blob/38f766ae0e4b80d452a3825c42a7ac9fee965790/cli/command/swarm/unlock.go#L47-L49

Should we have a single function to perform this check? (Not sure which one of the above is best :smile:)